test/e2e/sinkprefix_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or im
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+	"knative.dev/client/pkg/util"
+)
+
+const (
+	KnConfigContent string = `sink:
+- group: serving.knative.dev
+  prefix: hello
+  resource: services
+  version: v1`
+)
+
+type sinkprefixTestConfig struct {
+	knConfigDir  string
+	knConfigPath string
+}
+
+func (tc *sinkprefixTestConfig) setup() error {
+	var err error
+	tc.knConfigDir, err = ioutil.TempDir("", "kn1-config")
+	if err != nil {
+		return err
+	}
+	tc.knConfigPath, err = createPluginFile("config.yaml", KnConfigContent, tc.knConfigDir, FileModeReadWrite)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (tc *sinkprefixTestConfig) teardown() {
+	os.RemoveAll(tc.knConfigDir)
+}
+
+func TestSinkPrefixConfig(t *testing.T) {
+	t.Parallel()
+	test, err := NewE2eTest()
+	assert.NilError(t, err)
+	defer func() {
+		assert.NilError(t, test.Teardown())
+	}()
+
+	r := NewKnRunResultCollector(t)
+	defer r.DumpIfFailed()
+
+	tc := sinkprefixTestConfig{}
+	assert.NilError(t, tc.setup())
+	defer tc.teardown()
+
+	t.Log("Creating a testservice")
+	test.serviceCreate(t, r, "testsvc0")
+	t.Log("create cronJob sources with a sink to hello:testsvc0")
+	test.cronJobSourceCreateWithConfig(t, r, "testcronjobsource0", "* * * * */1", "ping", "hello:testsvc0", tc.knConfigPath)
+
+	jpSinkRefNameInSpec := "jsonpath={.spec.sink.ref.name}"
+	out, err := test.getResourceFieldsWithJSONPath("cronjobsource", "testcronjobsource0", jpSinkRefNameInSpec)
+	assert.NilError(t, err)
+	assert.Equal(t, out, "testsvc0")
+
+	t.Log("delete cronJob sources")
+	test.cronJobSourceDelete(t, r, "testcronjobsource0")
+}
+
+func (test *e2eTest) cronJobSourceCreateWithConfig(t *testing.T, r *KnRunResultCollector, sourceName string, schedule string, data string, sink string, config string) {
```

Comment:
Since CronJob source is deprecated and will be removed, I'd defer to use it for any *new* work/tests. As we'll need to re-work this later. Fine for now, but for any subsequent tests, lets use other sources.